webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -249,6 +249,19 @@ func (v validator) validateBootstrap(
 			allErrs = append(allErrs, field.TypeInvalid(p, vAppConfig,
 				"properties and rawProperties are mutually exclusive"))
 		}
+
+		if properties := vAppConfig.Properties; len(properties) != 0 {
+			for _, property := range properties {
+				if key := property.Key; key == "" {
```

Comment:
Nit: I don't think this scope `key` var add any clarity since we know it is `""` in the if body.